third-party/realdds/include/realdds/dds-device.h

```diff
@@ -48,22 +47,22 @@ class dds_device
 
     //----------- below this line, a device must be running!
 
-    size_t num_of_sensors() const;
+    size_t number_of_streams() const;
 
-    size_t foreach_sensor( std::function< void( size_t sensor_index, const std::string & name ) > fn ) const;
+    size_t foreach_stream( std::function< void( std::shared_ptr< dds_stream > stream ) > fn ) const;
 
-    size_t foreach_video_profile( size_t sensor_index, std::function< void( const rs2_video_stream& profile, bool def_prof ) > fn ) const;
-    size_t foreach_motion_profile( size_t sensor_index, std::function< void( const rs2_motion_stream& profile, bool def_prof ) > fn ) const;
-
-    void sensor_open( size_t sensor_index, const std::vector< rs2_video_stream > & profiles );
-    void sensor_close( size_t sensor_index );
+    void open( const std::vector< dds_video_stream::profile > & profiles );
+    void open( const std::vector< dds_motion_stream::profile > & profiles);
+    void close( const std::vector< std::pair< int16_t, int8_t > > & stream_uids );
 
 private:
     class impl;
     std::shared_ptr< impl > _impl;
 
     // Ctor is private: use find() or create() instead. Same for dtor -- it should be automatic
     dds_device( std::shared_ptr< impl > );
+
+    static std::shared_ptr< dds_device > find_internal( dds_guid const & guid );
```

Comment:
Please add some comments about this function, what does it do? when should we use it?

Reply:
Done